README.md
## Damascus v0

sprites/changelog.txt
Change log
========================


V3.2 [9-04-2023]
----------------
Updated the floors to have half for each side

V3.1 [27-02-2023]
----------------
Changed the characters so they are easier to integrate in a game

V3.0 [27-02-2023]
----------------
Revamped the characters
* New shadows and dust specs
* New clothing items
* New hairstyles
* Improved animations
* Added premade characters

V2.07 [13-12-2023]
----------------
Revamped the terrains

V2.06 [16-10-2022]
----------------
Added fish seller stall
Added fish seller shop
Added premade pink house (paid version only)
Added premade wooden cabin (paid version only)

V2.05 [09-10-2022]
----------------
Added ore seller stall
Added colours for animated potion house (paid version only)


V2.04 [06-10-2022]
----------------
Added new potion stall


V2.03 [03-10-2022]
----------------
Added new potion shop
Added potion shop animation (paid version only)


V2.02 [01-10-2022]
----------------
Added new marketplace building (paid version only)
Added new marketplace stall

V2.01 [29-09-2022]
----------------
Improved the housing system
New Wall colours
New Roof colours

V2.00 [18-09-2022]
----------------
Terrain revamp. Added four season to pack (paid version only)
Added rock cliffs.
Added seasonal items (paid version only)


V1.20 [6-08-2022]
----------------
Added the bushes destruction animations 

V1.19 [6-08-2022]
----------------
Added the bushes
Separated shadows from assets on outside assets

V1.18 [5-08-2022]
----------------
Added the squared terrain


V1.17 [27-07-2022]
----------------
Reconfigured the assets in to folders for each of them

V1.16 [26-06-2022]
----------------
Added children clothing

V1.15 [19-06-2022]
----------------
Added children hair

V1.14 [6-04-2022]
----------------
Added children characters
Basic child clothes

V1.13 [29-04-2022]
----------------
Revamped the kitchen cabinets
Added a new style
Added new fridges

V1.12 [15-04-2022]
----------------
Added new tables
Added twin beds
Added table cloths

V1.11 [11-03-2022]
----------------
Added a new Skin Tone
Added colours to all hairs (red, blonde, Browns, Black, Purple)
Added a new Balding Gray hair

V1.10 [04-03-2022]
----------------
Added street lamps
Added street signs

V1.9 [25-02-2022]
----------------
Added Opening doors inside
Added trash cans
Added standup mailboxes
Added Trees inside pens
Added benches without back support
Improved cobblestones
Added new cobblestones
Added Wood Signs
Added empty boxes
Added new flower


V1.8 [18-02-2022]
----------------
Added new doors
Added new above door asset
Removed Aseprite files to commercial version

V1.7 [11-02-2022]
----------------
Added lamps
Added mailboxes
Added a town statue
Added benches
Added crates

V1.6.2 [07-02-2022]
----------------
Fixed export of khakis pants

V1.6.1 [05-02-2022]
----------------
Fixed colours and shadows on the housing tiles
Fixed wrong left side for tShirts and shoes

V1.6 [02-02-2022]
----------------
Added Kitchen Sink
Added Kitchen cabinets
Added Kitchen cuboards
Added Table cloths
Added Fridges

V1.5 [26-01-2022]
----------------
Added wooden chairs & tables
Added bookcases and books
Added Coaches
Updated wall system
Added Fireplace (no animation)

V1.4 [19-01-2022]
----------------
Added the new housing system
Added interior walls
Added new chimneys 
Added smoke for chimneys

V1.3 [12-01-2022]
----------------
Updated the character in 4 directions
Merged the hands into one unique character layer
Updates all clothing items to reflect these new directions
Updates all hairs items to reflect these new directions

V1.2 [07-01-2022]
----------------
New Assets:
* Animated Ground(dirt) shore
* Middle Trees
    * Pointy
    * Cilinder
    * Round
* Big Tree (Round)
* Fountains:
    * Unstable water movement
    * Stable water

Format:
* Updated tiles sheet to be easier to pick tiles

--

V1.1 [04-01-2022]
-----------------
* Added sideways for water animation
* Corrected brown terrain stairs
* added horizontal and vertical pathways

--

V1.0 [03-01-2022]
-----------------
* Released v1.0


main.c
#include <gb/gb.h>
#include <stdio.h>
#include "Ecklin.c"
#include "GameCharacter.c"

struct GameCharacter ecklin;
UBYTE spriteSize = 8;

void performantdelay(UINT8 numloops)
{
  UINT8 i;
  for(i =0; i < numloops; i++) {
    wait_vbl_done();
  }
}

void moveGameCharacter(struct GameCharacter *character, UINT8 x, UINT8 y)
{
  move_sprite(character->spritids[0], x, y);
  move_sprite(character->spritids[1], x + spriteSize, y);
  move_sprite(character->spritids[2], x, y + spriteSize);
  move_sprite(character->spritids[3], x + spriteSize, y + spriteSize);
}

void setupEcklin()
{
  ecklin.x = 80;
  ecklin.y = 130;
  ecklin.width = 16;
  ecklin.height = 16;

  set_sprite_tile(0, 0);
  ecklin.spritids[0] = 0;
  set_sprite_tile(1, 1);
  ecklin.spritids[1] = 1;
  set_sprite_tile(2, 2);
  ecklin.spritids[2] = 2;
  set_sprite_tile(3, 3);
  ecklin.spritids[3] = 3;

  moveGameCharacter(&ecklin, ecklin.x, ecklin.y);
}

void main()
{
  set_sprite_data(0, 8, Ecklin);
  setupEcklin();
  SHOW_SPRITES;
  DISPLAY_ON;

  while (1)
  {
    if(joypad() & J_LEFT)
    {
      ecklin.x -= 2;
      moveGameCharacter(&ecklin, ecklin.x, ecklin.y);
    }
    if(joypad() & J_RIGHT)
    {
      ecklin.x += 2;
      moveGameCharacter(&ecklin, ecklin.x, ecklin.y);
    }

     if(joypad() & J_UP)
    {
      ecklin.y -= 2;
      moveGameCharacter(&ecklin, ecklin.x, ecklin.y);
    }

     if(joypad() & J_DOWN)
    {
      ecklin.y += 2;
      moveGameCharacter(&ecklin, ecklin.x, ecklin.y);
    }
    performantdelay(5);
  }
}

main.lua
function love.load()
    anim8 = require 'libraries/anim8'
    camera = require 'libraries/camera'
    sti = require 'libraries/sti'
    wf = require 'libraries/windfield'
    world = wf.newWorld(0, 0, false)
    gameMap = sti('maps/homebase.lua')
    cam = camera()
    cam.scale = 3
    love.graphics.setDefaultFilter("nearest", "nearest")

    player = {}
    player.collider = world:newRectangleCollider(16, 16, 16, 16)
    player.collider:setFixedRotation(true)
    player.x = 16
    player.y = 16
    player.speed = 100 -- Adjust this value to control player speed
    player.spriteSheet = love.graphics.newImage('sprites/myles.png')
    player.grid = anim8.newGrid(16, 16, player.spriteSheet:getWidth(), player.spriteSheet:getHeight())
    player.animations = {}
    player.animations.down = anim8.newAnimation(player.grid('1-3', 1), 0.2)
    player.animations.left = anim8.newAnimation(player.grid('1-2', 3), 0.3)
    player.animations.right = anim8.newAnimation(player.grid('1-2', 4), 0.3)
    player.animations.up = anim8.newAnimation(player.grid('1-3', 2), 0.3)

    player.anim = player.animations.left

    background = love.graphics.newImage('sprites/background.png')

    doorway = nil

    walls = {}
    if gameMap.layers["Building"] then
        for i, obj in pairs(gameMap.layers["Building"].objects) do
            local wall = world:newRectangleCollider(obj.x,obj.y, obj.width, obj.height)
            wall:setType('static')
            table.insert(walls, wall)
        end
    end
    if gameMap.layers["Entrance"] then
        for i, obj in pairs(gameMap.layers["Entrance"].objects) do
            doorway = world:newRectangleCollider(obj.x,obj.y, obj.width, obj.height)
            doorway:setType('static')
        end
    end 

end

function love.update(dt)
    local isMoving = false
    local vx = 0
    local vy = 0

    if love.keyboard.isDown("right") then
        vx = player.speed
        player.anim = player.animations.right
        isMoving = true
    end
    if love.keyboard.isDown("left") then
        vx = -player.speed
        player.anim = player.animations.left
        isMoving = true
    end
    if love.keyboard.isDown("up") then
        vy = -player.speed
        player.anim = player.animations.up
        isMoving = true
    end
    if love.keyboard.isDown("down") then
        vy = player.speed
        player.anim = player.animations.down
        isMoving = true
    end

    player.collider:setLinearVelocity(vx, vy)

    if isMoving == false then
        player.anim:gotoFrame(1)
    end

    if player.collider:isTouching(doorway:getBody()) then
        local doorX, doorY = doorway:getPosition()
        player.x = doorX
        player.y = doorY

        gameMap = sti('maps/homebase_inside.lua')
        cam.x = player.x - love.graphics.getWidth() / 2
cam.y = player.y - love.graphics.getHeight() / 2
cam:zoomTo(1)
    end

    world:update(dt)
    player.x = player.collider:getX()
    player.y = player.collider:getY()
    player.anim:update(dt)
    cam:lookAt(player.x, player.y)

    local w = love.graphics.getWidth()
    local h = love.graphics.getHeight()

    if cam.x < w / 6 then
        cam.x = w / 6
    end
    if cam.y < h / 6 then
        cam.y = h / 6
    end

    local mapW = gameMap.width * gameMap.tilewidth
    local mapH = gameMap.height * gameMap.tileheight

    if cam.x > (mapW - w / 5) then
        cam.x = (mapW - w / 5)
    end

    if cam.y > (mapH - h / 5) then
        cam.y = (mapH - h / 5)
    end
end

function love.draw()
    cam:attach()
    love.graphics.draw(background)
    gameMap:drawLayer(gameMap.layers["Tile Layer 1"])
    gameMap:drawLayer(gameMap.layers["Tile Layer 2"])
    player.anim:draw(player.spriteSheet, player.x, player.y, nil, nil, nil, 8, 8)
    world:draw()
    cam:detach()
end

maps/homebase.lua
return {
  version = "1.10",
  luaversion = "5.1",
  tiledversion = "1.10.1",
  class = "",
  orientation = "orthogonal",
  renderorder = "right-down",
  width = 30,
  height = 30,
  tilewidth = 16,
  tileheight = 16,
  nextlayerid = 5,
  nextobjectid = 8,
  properties = {},
  tilesets = {
    {
      name = "terrain",
      firstgid = 1,
      class = "",
      tilewidth = 16,
      tileheight = 16,
      spacing = 0,
      margin = 16,
      columns = 63,
      image = "../sprites/assets1/tileset/terrain/smallburg_spring_terrain.png",
      imagewidth = 1024,
      imageheight = 399,
      objectalignment = "unspecified",
      tilerendersize = "tile",
      fillmode = "stretch",
      tileoffset = {
        x = 0,
        y = 0
      },
      grid = {
        orientation = "orthogonal",
        width = 16,
        height = 16
      },
      properties = {},
      wangsets = {},
      tilecount = 1449,
      tiles = {}
    },
    {
      name = "doors",
      firstgid = 1450,
      class = "",
      tilewidth = 16,
      tileheight = 16,
      spacing = 0,
      margin = 16,
      columns = 46,
      image = "../sprites/assets1/housing/house_doors.png",
      imagewidth = 752,
      imageheight = 512,
      objectalignment = "unspecified",
      tilerendersize = "tile",
      fillmode = "stretch",
      tileoffset = {
        x = 0,
        y = 0
      },
      grid = {
        orientation = "orthogonal",
        width = 16,
        height = 16
      },
      properties = {},
      wangsets = {},
      tilecount = 1426,
      tiles = {}
    },
    {
      name = "housing_system_roofs",
      firstgid = 2876,
      class = "",
      tilewidth = 16,
      tileheight = 16,
      spacing = 0,
      margin = 16,
      columns = 74,
      image = "../sprites/assets1/housing/housing_system_roofs.png",
      imagewidth = 1200,
      imageheight = 704,
      transparentcolor = "#8fdf5c",
      objectalignment = "unspecified",
      tilerendersize = "tile",
      fillmode = "stretch",
      tileoffset = {
        x = 0,
        y = 0
      },
      grid = {
        orientation = "orthogonal",
        width = 16,
        height = 16
      },
      properties = {},
      wangsets = {},
      tilecount = 3182,
      tiles = {}
    },
    {
      name = "house_building_assets",
      firstgid = 6058,
      class = "",
      tilewidth = 16,
      tileheight = 16,
      spacing = 0,
      margin = 16,
      columns = 74,
      image = "../sprites/assets1/housing/house_building_assets.png",
      imagewidth = 1200,
      imageheight = 704,
      transparentcolor = "#8fdf5c",
      objectalignment = "unspecified",
      tilerendersize = "tile",
      fillmode = "stretch",
      tileoffset = {
        x = 0,
        y = 0
      },
      grid = {
        orientation = "orthogonal",
        width = 16,
        height = 16
      },
      properties = {
        ["trans"] = "#ff8fdf5c"
      },
      wangsets = {},
      tilecount = 3182,
      tiles = {}
    }
  },
  layers = {
    {
      type = "tilelayer",
      x = 0,
      y = 0,
      width = 30,
      height = 30,
      id = 1,
      name = "Tile Layer 1",
      class = "",
      visible = true,
      opacity = 1,
      offsetx = 0,
      offsety = 0,
      parallaxx = 1,
      parallaxy = 1,
      properties = {},
      encoding = "lua",
      data = {
        113, 113, 931, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 65,
        113, 113, 931, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 65,
        113, 113, 931, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 65,
        113, 113, 931, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 1041, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113,
        113, 113, 931, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 1102, 1103, 1104, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113,
        113, 113, 931, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 1166, 1167, 1168, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113,
        113, 113, 931, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 1229, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113,
        113, 113, 931, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113,
        113, 113, 931, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113,
        113, 113, 931, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113,
        113, 113, 931, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113,
        113, 113, 931, 113, 113, 113, 113, 113, 113, 113, 113, 113, 6872, 6873, 6874, 6875, 6876, 6877, 6878, 6879, 6880, 113, 113, 113, 113, 113, 113, 113, 113, 113,
        113, 113, 931, 113, 113, 113, 113, 113, 113, 113, 113, 113, 6946, 6947, 6948, 6949, 6950, 6951, 6952, 6953, 6954, 113, 113, 113, 113, 113, 113, 113, 113, 113,
        113, 113, 931, 113, 113, 113, 113, 113, 113, 113, 113, 113, 7020, 7021, 7022, 7023, 7024, 7025, 7026, 7027, 7028, 113, 113, 113, 113, 113, 113, 113, 113, 113,
        113, 113, 931, 113, 113, 113, 113, 113, 113, 113, 113, 113, 7094, 7095, 7096, 7097, 7098, 7099, 7100, 7101, 7102, 113, 113, 113, 113, 113, 113, 113, 113, 113,
        113, 113, 931, 113, 113, 113, 113, 113, 113, 113, 113, 113, 7168, 7169, 7170, 7171, 7172, 7173, 7174, 7175, 7176, 113, 113, 113, 113, 113, 113, 113, 113, 113,
        113, 113, 931, 113, 113, 113, 113, 113, 113, 113, 113, 113, 7242, 7243, 7244, 7245, 7246, 7247, 7248, 7249, 7250, 113, 113, 113, 113, 113, 113, 113, 113, 113,
        113, 113, 931, 113, 113, 113, 113, 113, 113, 113, 113, 113, 7316, 7317, 7318, 7319, 7320, 7321, 7322, 7323, 7324, 113, 113, 113, 113, 113, 113, 113, 113, 113,
        113, 113, 931, 113, 113, 113, 113, 113, 113, 113, 113, 113, 7390, 7391, 7392, 7393, 0, 7395, 7396, 7397, 7398, 113, 113, 113, 113, 113, 113, 113, 113, 113,
        113, 113, 931, 113, 113, 113, 113, 113, 113, 113, 113, 113, 7464, 7465, 7466, 7467, 7468, 7469, 7470, 7471, 7472, 113, 113, 113, 113, 113, 113, 113, 113, 113,
        113, 113, 931, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 931, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113,
        113, 113, 931, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 931, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113,
        113, 113, 991, 803, 803, 803, 803, 803, 803, 803, 803, 803, 803, 803, 803, 803, 993, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113,
        113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113,
        113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113,
        113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113,
        113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113,
        113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113,
        113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113,
        113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113, 113
      }
    },
    {
      type = "objectgroup",
      draworder = "topdown",
      id = 4,
      name = "Entrance",
      class = "",
      visible = true,
      opacity = 1,
      offsetx = 0,
      offsety = 0,
      parallaxx = 1,
      parallaxy = 1,
      properties = {},
      objects = {
        {
          id = 7,
          name = "",
          type = "",
          shape = "rectangle",
          x = 256.667,
          y = 288.667,
          width = 15,
          height = 30.3333,
          rotation = 0,
          visible = true,
          properties = {}
        }
      }
    },
    {
      type = "objectgroup",
      draworder = "topdown",
      id = 3,
      name = "Building",
      class = "",
      visible = true,
      opacity = 1,
      offsetx = 0,
      offsety = 0,
      parallaxx = 1,
      parallaxy = 1,
      properties = {},
      objects = {
        {
          id = 1,
          name = "",
          type = "",
          shape = "rectangle",
          x = 194.667,
          y = 160.667,
          width = 138.667,
          height = 126.667,
          rotation = 0,
          visible = true,
          properties = {}
        },
        {
          id = 3,
          name = "",
          type = "",
          shape = "rectangle",
          x = 278,
          y = 288.667,
          width = 54.6667,
          height = 31.3333,
          rotation = 0,
          visible = true,
          properties = {}
        },
        {
          id = 5,
          name = "",
          type = "",
          shape = "rectangle",
          x = 197,
          y = 286.667,
          width = 53.6667,
          height = 33,
          rotation = 0,
          visible = true,
          properties = {}
        },
        {
          id = 6,
          name = "",
          type = "",
          shape = "rectangle",
          x = 245,
          y = 150.333,
          width = 39,
          height = 10,
          rotation = 0,
          visible = true,
          properties = {}
        }
      }
    },
    {
      type = "tilelayer",
      x = 0,
      y = 0,
      width = 30,
      height = 30,
      id = 2,
      name = "Tile Layer 2",
      class = "",
      visible = true,
      opacity = 1,
      offsetx = 0,
      offsety = 0,
      parallaxx = 1,
      parallaxy = 1,
      properties = {},
      encoding = "lua",
      data = {
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 3693, 3694, 3695, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 3764, 3765, 3766, 3767, 3768, 3769, 3770, 3771, 3772, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 3838, 3839, 3840, 3841, 3842, 3843, 3844, 3845, 3846, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 3912, 3913, 3914, 3915, 3916, 3917, 3918, 3919, 3920, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 3986, 3987, 3988, 3989, 3990, 3991, 3992, 3993, 3994, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 4060, 4061, 4062, 4063, 4064, 4065, 4066, 4067, 4068, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 4134, 4135, 4136, 4137, 4138, 4139, 4140, 4141, 4142, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 4208, 4209, 4210, 4211, 4212, 4213, 4214, 4215, 4216, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 4282, 4283, 4284, 4285, 4286, 4287, 4288, 4289, 4290, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 4356, 4357, 4358, 4359, 2278, 4361, 4362, 4363, 4364, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 931, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0
      }
    }
  }
}
